zio-http/shared/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -712,68 +712,82 @@ object JsonSchema {
         fromZSchema(schema, refType)
       case Schema.Primitive(standardType, annotations)                                            =>
         standardType match {
-          case StandardType.UnitType           => JsonSchema.Null
-          case StandardType.StringType         =>
-            JsonSchema.String.fromValidation(
+          case StandardType.UnitType                               => JsonSchema.Null
+          case StandardType.StringType | StandardType.DayOfWeekType | StandardType.MonthType |
+              StandardType.MonthDayType | StandardType.YearType | StandardType.YearMonthType | StandardType.ZoneIdType |
+              StandardType.ZoneOffsetType | StandardType.LocalDateTimeType | StandardType.LocalTimeType |
+              StandardType.CurrencyType | StandardType.BinaryType | StandardType.CharType =>
+            JsonSchema.String.fromValidationAndFormat(
               annotations.collect { case zio.schema.annotation.validate(v) => v }.headOption,
+              annotations.collect { case format(stringFormat) => stringFormat }.headOption,
```

Comment:
I don't think an annotation is the right solution.
Especially not one that lives in this file.

I think right solution is to have a mapping of `StandardType` to a format.
If this is not precise enough, then we need something in zio schema to make it configurable